framework/Models/PostProcessors/BasicStatistics.py

```diff
@@ -957,6 +988,46 @@ def __runLocal(self, inputData):
         percentileSet = percentileSet.rename({'quantile':'percent'})
       calculations[metric] = percentileSet
 
+      # because percentile is different, calculate standard error here
+      self.raiseADebug('Starting calculate standard error on "'+metric+'"...')
+      percentileSteSet = xr.Dataset()
+      calculatedPercentiles = calculations[metric]
+      relWeight = pbWeights[list(needed[metric]['targets'])]
+      for target in needed[metric]['targets']:
+        targWeight = relWeight[target].values
+        en = targWeight.sum()**2/np.sum(targWeight**2)
+        targDa = dataSet[target]
+        if self.pivotParameter in targDa.sizes.keys():
+          percentileSte = []
+          for pct in percent:
+            subPercentileSte = []
+            factor = np.sqrt(pct*(1.0 - pct)/en)
+            for label, group in targDa.groupby(self.pivotParameter):
+              if group.values.min() == group.values.max():
+                # all values are the same
+                subPercentileSte.append(0.0)
+              else:
+                # get KDE
+                kde = stats.gaussian_kde(group.values, weights=targWeight)
+                val = calculatedPercentiles[target].sel(**{'percent': pct, self.pivotParameter: label}).values
+                subPercentileSte.append(factor/kde(val)[0])
+            percentileSte.append(subPercentileSte)
+          da = xr.DataArray(percentileSte, dims=('percent', self.pivotParameter), coords={'percent': percent, self.pivotParameter: self.pivotValue})
+          percentileSteSet[target] = da
+        else:
+          calcPercentiles = calculatedPercentiles[target]
+          if targDa.values.min() == targDa.values.max():
+            # distribution is a delta function, so no KDE construction
+            percentileSte = list(np.zeros(calcPercentiles.shape))
+          else:
+            # get KDE
+            kde = stats.gaussian_kde(targDa.values, weights=targWeight)
+            factor = np.sqrt(np.array(percent)*(1.0 - np.array(percent))/en)
+            percentileSte = list(factor/kde(calcPercentiles.values))
```

Comment:
@dgarrett622 I have a question regarding the calculation of standard error for percentile, from the reference (see the following snapshot)
![image](https://user-images.githubusercontent.com/7321071/154192571-8a9f2c79-6e9f-48ac-959c-c4c9c1effee4.png)

The standard normal distribution is used instead of KDE of real data, could you double check with the reference, and let me know what do you think?

Reply:
@wangcj05 That reference gives an equation for the asymptotic standard error estimate of a percentile assuming the random variable belongs to a normal distribution. There is a more general form of this asymptotic estimate that can be found in various references such as:

[http://www.medicine.mcgill.ca/epidemiology/hanley/bios601/DescriptiveStatistics/Var(percentile).pdf](http://www.medicine.mcgill.ca/epidemiology/hanley/bios601/DescriptiveStatistics/Var(percentile).pdf)

Here is a snapshot of the more general asymptotic estimate from the reference above which I implemented:
<img width="447" alt="percentile_ste" src="https://user-images.githubusercontent.com/19575707/154278689-0c83851f-94ec-4687-bf5b-a4ef8a076e05.PNG">

The KDE tries to get the real distribution from the data without making the assumption that the data are normally distributed. The percentile, its standard error, and the KDE perform better with more samples, but that was always going to be a limitation when requesting a percentile.

Reply:
@wangcj05 The two equations are not equivalent. Using a Gaussian kernel is not the same as fitting a Gaussian to the data. The Gaussian kernel is like smoothing or interpolating to fit a distribution to the real data. Here is an example of using a Gaussian kernel to find a distribution that is definitely not Gaussian:

[https://gsalvatovallverdu.gitlab.io/python/kernel_density_estimation/](https://gsalvatovallverdu.gitlab.io/python/kernel_density_estimation/)

With the distribution fit to the data becoming:
<img width="429" alt="gaussian_kde" src="https://user-images.githubusercontent.com/19575707/154302645-3a9a48e5-ac93-49b6-9029-2107084f4427.PNG">

The number of samples is an issue here, both in accuracy and computational cost. Determining the percentile itself requires many samples, especially if the requested percentile is in the tail of the distribution. The more samples you take, the more time it takes to compute the percentile since we do sorting on the data itself. The limitation in the paper applies to the calculation of the percentile itself and the standard error calculation, whether the standard normal formulation or the KDE method is used since it is based on number of samples. However, the standard normal formulation assumes that the random variable is distributed normally while the KDE method makes no assumption about the distribution.

An alternate method would be to use bootstrapped samples, but again there are concerns about computational time with that method.